### ISSUE TITLE
Controller 8bitdo SN30 not responding with Chromecast with google Tv

### ISSUE THREAD

**Issue by molle83** [Swordfish90/Lemuroid#166]
Hi.
I tried to play with my controller 8bitdo SN30 on my Chromecast with google Tv but non of the keys or buttons will respond, I can navigate with the controller in the chromecast menus and in Lemudroid app but when i start a game the buttons no longer  responding.
I tried the lemudroid on an Meecool M8S Pro L with AndroidTV 7 and works in the games flawlessly.
I tried with another app on the chromecast Snes9x EX+ and the 8bitdo works in the games flawlessly.
 
I also tried the controller on my galaxy tab 4 with Lemuroid and there the 8bitdo also works in the games flawlessly.
 
So its something with chromecast?
I have assigned my buttons manually because the where assigned wrong, example button a was assigned button b, the buttons L2, R2 etc that i dont have on sn30 i also set to unassigne.
The latest version 1.8.1 on Lemudroid.
 
I have the newest firmware 6.14 on the controller.

**Comment by Swordfish90** [comment]
Hello again, and thank you for reporting the issue also here. I've got a tentative fix for your issue, can you please install this apk on your chromecast and let me know if it fixes your issues?

https://drive.google.com/file/d/1oo98FCmj4U8Rn1fmmJvj-VKbFJqilVPx/view?usp=sharing

It can be installed alongside the standard application and has "LemuroidDebug" as name. It should filter the virtual-search gamepad.

**Comment by molle83** [comment]
Hi. Sprry but didnt help./Emil
-------- Originalmeddelande --------Från: Filippo Scognamiglio <notifications@github.com> Datum: 2021-01-11  19:21  (GMT+01:00) Till: Swordfish90/Lemuroid <Lemuroid@noreply.github.com> Kopia: molle83 <emil@tyresmo.se>, Author <author@noreply.github.com> Ämne: Re: [Swordfish90/Lemuroid] Controller 8bitdo SN30 not responding with Chromecast with google Tv (#166) 
Hello again, and thank you for reporting the issue also here. I've got a tentative fix for your issue, can you please install this apk on your chromecast and let me know if it fixes your issues?
https://drive.google.com/file/d/1oo98FCmj4U8Rn1fmmJvj-VKbFJqilVPx/view?usp=sharing
It can be installed alongside the standard application and has "LemuroidDebug" as name. It should filter the virtual-search gamepad.

**Comment by Swordfish90** [comment]
One would assume that "virtual-search" would be a virtual device, but apparently I was wrong... Here I can find another build that basically blacklists the device.

https://drive.google.com/file/d/11jgmWcykS0_I7uLFvZuErdJllne3pBNz/view?usp=sharing

Please give it another go when you have some time, and I'll include it in the next release.

**Comment by molle83** [comment]
Hi!

YES it worked. GREAT!

 

One other thing, it came up an text when the game starts something like “Access settings press L3+R3”, I don’t have these buttons on the SN30. Is this like the consoles “reset” button?

 

/Emil

**Comment by Swordfish90** [comment]
It appears that some 8bitdo controllers expose keys which are basically not there, I've been able to reproduce it with my Zero 2, which also exposes L3 and R3. There isn't a real fix, but I just pushed a workaround which will allow you to customize the keys for the Game Menu to either L3 + R3 or Select + Start. I'll be available with Lemuroid 1.9.0.

**Comment by molle83** [comment]
Ok great!/Emil
-------- Originalmeddelande --------Från: Filippo Scognamiglio <notifications@github.com> Datum: 2021-01-14  02:00  (GMT+01:00) Till: Swordfish90/Lemuroid <Lemuroid@noreply.github.com> Kopia: molle83 <emil@tyresmo.se>, Author <author@noreply.github.com> Ämne: Re: [Swordfish90/Lemuroid] Controller 8bitdo SN30 not responding with Chromecast with google Tv (#166) 
It appears that some 8bitdo controllers expose keys which are basically not there, I've been able to reproduce it with my Zero 2, which also exposes L3 and R3. There isn't a real fix, but I just pushed a workaround which will allow you to customize the keys for the Game Menu to either L3 + R3 or Select + Start. I'll be available with Lemuroid 1.9.0.
